Documentation/network/concepts/ipam/cluster-pool.rst

```diff
@@ -45,6 +45,13 @@ Configuration
 For a practical tutorial on how to enable this mode in Cilium, see
 :ref:`gsg_ipam_crd_cluster_pool`.
 
+.. note::
+    An existing element of the ``clusterPoolIPv4PodCIDRList`` list shouldn't be removed or changed. 
+    Doing so will cause unexpected behaviour. If you face a situation where the pool is exhausted,
+    add a new element to the list. The minimum mask length is ``/30``, but a mask of at least ``/29`` 
+    is recommended. The reason for this is that the allocator will reserve 2 IPs per CIDR block for 
+    the network and broadcast addresses. Changing ``clusterPoolIPv4MaskSize`` is also not supported.
```

Comment:
nit: "not supported" gives the impression that for the rest of the project, there is some sort of formal support agreement. However there is not, all of Cilium is provided `on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND` as per the license.

Perhaps @zacharysarah has some sort of suggestion as to how to appropriately phrase that this sort of change in configuration is not tested / not validated by the project and hence the risk is entirely on the user if they decide to make such changes (but also we would happily accept patches to improve this!).

Reply:
Same "not supported" wording shows up in `index.rst` as well, whatever we decide this feedback should apply to that case as well.

Reply:
I think it's better to omit any mention of support for the reasons @joestringer mentions. If we need to give advice about use cases, then let's speak to the specific use case. For example: "Changing ``value`` isn't recommended for production environments." We could also just say: "Changing ``value`` isn't recommended."

Reply:
Maybe to clarify, "not supported" in this context here means "it will not work"/"will break your cluster". So it's not only not recommended, but something that our code simply cannot deal with.

 Ideally of course, we would enforce this, and eventually we will probably have the ability to do prevent the user from doing this (I think Dario has some ideas), but for now we want to at least document it.

Reply:
My understanding of the behaviour is that nothing will happen if you change the IPAM mode. Is my understanding wrong?

Reply:
I don't have specific knowledge in this regard, but I was teeing off @gandro's comment above that it will "break your cluster" in a possibly recoverable way. My guess is that if you reconfigure then upon Cilium restart it will try to re-allocate the previously-used IPs from the new pool and that may not be possible, so then I don't know if Cilium will fail restore of existing workloads or let them persist with IPs that may or may not be usable.

Reply:
The "is not possible" statement was with regards to the "changing `clusterPoolIPv4MaskSize`" sentence in the highlighted patch line. Doing this will cause the operator to reject any existing pod CIDRs on a node and allocate new pod CIDRs with the new mask size. I also think this means that some of the IP space of a node with the old mask size can be reused by a different node, which then leads to duplicate IPs on pods.

Changing the IPAM mode in a cluster with running pods will cause agents and operator to ignore the old IPAM IPs and start allocating new ones. Existing pods still have IPs from the old IPAM mode and likely be in a defunct state (e.g. routes might not point to them anymore, endpoint restoration will not work etc, the list here is rather long).

Reply:
:+1: thanks for the details. To summarize, that sounds a bit more like:

> Changing the IPAM mode of an existing cluster is strongly discouraged as it may cause persistent connectivity disruption for existing workloads.